File: frontend/src/lib/maps.ts
import type { Place } from "@/types/place";

export function getGoogleMapsUrl(place: Place): string {
  const location =
    place.address &&
    place.address !== "Address unavailable"
      ? `${place.name}, ${place.address}, ${place.city}`
      : `${place.name}, ${place.city}`;

  const parameters = new URLSearchParams({
    api: "1",
    query: location,
  });

  return `https://www.google.com/maps/search/?${parameters.toString()}`;
}

export function getGoogleDirectionsUrl(
  place: Place,
): string {
  const parameters = new URLSearchParams({
    api: "1",
    destination: `${place.latitude},${place.longitude}`,
    dir_action: "navigate",
  });

  return `https://www.google.com/maps/dir/?${parameters.toString()}`;
}

File: backend/app/core/maps.py
from urllib.parse import urlencode


GOOGLE_MAPS_TRANSIT_DISCLAIMER = (
    "Open Google Maps for current public-transport directions. Routes, "
    "departure times, disruptions, and availability may change; verify "
    "the latest information before travelling."
)


def get_google_maps_transit_url(latitude: float, longitude: float) -> str:
    """Build a key-free Google Maps transit directions URL."""

    parameters = urlencode(
        {
            "api": "1",
            "destination": f"{latitude},{longitude}",
            "travelmode": "transit",
            "dir_action": "navigate",
        }
    )
    return f"https://www.google.com/maps/dir/?{parameters}"
